sendermode.go

```diff
@@ -154,36 +157,62 @@ func (fm *Frontman) sendResultsChanToHubQueue(resultsChan *chan Result) error {
 				shouldReturn = true
 			} else {
 				results = append(results, res)
-				if len(results) < fm.Config.QueueSenderBatchSize && len(*resultsChan) > 0 {
+				if len(results) < fm.Config.SenderBatchSize && len(*resultsChan) > 0 {
 					continue
 				}
 			}
 		}
 
-		duration := time.Since(lastSentToHub)
-		if duration >= interval {
-
-			if len(results) >= fm.Config.QueueSenderBatchSize {
-				sendResults = results[0:fm.Config.QueueSenderBatchSize]
-				results = results[fm.Config.QueueSenderBatchSize:]
+		if time.Since(lastSentToHub) >= sendInterval {
+			if len(results) >= fm.Config.SenderBatchSize {
+				sendResults = results[0:fm.Config.SenderBatchSize]
+				results = results[fm.Config.SenderBatchSize:]
 			} else {
 				sendResults = results
 				results = nil
 			}
 
 			if len(sendResults) > 0 {
-				logrus.Infof("Send %d results to hub, %d results in queue", len(sendResults), len(results))
 				lastSentToHub = time.Now()
 				go func(r []Result) {
 					err := fm.postResultsToHub(r)
 					if err != nil {
+						if err == ErrorHubGeneral {
+							// If the hub doesn't respond with 2XX, the results remain in the queue.
+							results = append(results, sendResults...)
```

Comment:
it's a potential data race since results is written by go routine and read by the main process. The recommended way of communication with go routines are channels rather than shared variables